File: src/main/java/io/dddrive/search/service/api/SearchService.java

package io.dddrive.search.service.api;

import java.util.List;

import org.jooq.Condition;
import org.jooq.Field;

import io.crnk.core.queryspec.FilterSpec;
import io.dddrive.app.model.RequestContext;
import io.dddrive.ddd.model.Aggregate;
import io.dddrive.search.model.SearchResult;

public interface SearchService {

	/**
	 * Store the search texts and tokens.
	 *
	 * @param aggregate aggregate to store
	 * @param texts     list of texts to be stored
	 * @param tokens    list of tokens to be stored
	 */
	void storeSearch(Aggregate aggregate, List<String> texts, List<String> tokens);

	List<SearchResult> find(RequestContext requestCtx, String searchText, int maxResultSize);

	List<SearchResult> find(RequestContext requestCtx, List<String> itemTypes, String searchText, int maxResultSize);

	Condition searchFilter(Field<Integer> idField, FilterSpec filter);

}

File: fm-domain/src/main/java/io/zeitwert/dddrive/ddd/service/api/SearchService.kt
package io.zeitwert.dddrive.ddd.service.api

import dddrive.ddd.model.Aggregate
import io.zeitwert.app.model.SessionContext
import io.zeitwert.dddrive.ddd.model.SearchResult

interface SearchService {

	/**
	 * Store the search texts and tokens.
	 *
	 * @param aggregate aggregate to store
	 * @param texts     list of texts to be stored
	 * @param tokens    list of tokens to be stored
	 */
	fun storeSearch(
		aggregate: Aggregate,
		texts: List<String?>,
		tokens: List<String?>,
	)

	fun findOne(
		sessionContext: SessionContext,
		itemType: String,
		searchText: String,
	): SearchResult?

	fun find(
		sessionContext: SessionContext,
		searchText: String,
		maxResultSize: Int,
	): List<SearchResult>

	fun find(
		sessionContext: SessionContext,
		itemTypes: List<String>?,
		searchText: String,
		maxResultSize: Int,
	): List<SearchResult>

}
